src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -183,6 +193,12 @@ protected override void ApplyResponseChallenge()
                 State = OpenIdConnectAuthenticationDefaults.AuthenticationPropertiesKey + "=" + UrlEncoder.UrlEncode(Options.StateDataFormat.Protect(properties))
             };
 
+            // if response_type=code, nonce is not required.
+            if (Options.ResponseType.Equals(OpenIdConnectResponseTypes.Code))
```

Comment:
Need to check the value of OpenIdConnectMessage.ResponseType, since it can be set in RedirectToIdentityProvider later. So any micro changes should be based off of the final request sent to the IDP.